employee/sealed.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Starting from a `read_only` struct, this program
+ * restores full privileges using a previous `sealed`
+ * capabilty.
+ */
+
+#include "include/employee.h"
+#include <assert.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/sysctl.h>
+
+
+
+int main()
+{
+	struct employee employee = {0, "Good", "Employee", 55000};
+	struct employee *ro_employee = set_read_only(&employee);
+	printf("*** `ro_employee` is read-only ***");
+	pp_cap(ro_employee);
+	print_details(ro_employee);
+	// Data to update the employee
+	struct employee updated_employee = {1, "Better", "Unsealed", 250000};
+	struct employee *updated_employee_ptr = &updated_employee;
+	// Pointers for the sealed and unsealed capability
+	struct employee *sealed_employee;
+	struct employee *unsealed_employee;
+	void *__capability sealcap;
+	size_t sealcap_size;
+
+	// Request special capability `sealcap` from the operating system
+	// in order to use it as key to seal `updated_employee_ptr`
+	sealcap_size = sizeof(sealcap);
+	if (sysctlbyname("security.cheri.sealcap", &sealcap, &sealcap_size, NULL, 0) < 0)
+		error("Something is wrong. Cannot get `securiy.cheri.sealcap`.");
+	assert(cheri_perms_get(sealcap) & CHERI_PERM_SEAL);
+	// Explicitly seal `updated_employee_ptr`
+	sealed_employee = (struct employee *) cheri_seal(updated_employee_ptr, sealcap);
+	assert(cheri_is_sealed(sealed_employee));
+	// Explicitly unseal `updated_employee_ptr`
+	unsealed_employee = (struct employee *) cheri_unseal(sealed_employee, sealcap);
+	assert(cheri_is_unsealed(unsealed_employee));
+	ro_employee = unsealed_employee;
+	printf("*** Using a previous sealed capability we restore read and write permissions ***\n");
+	printf("*** Note: this operation is completely transparent to the user               ***\n");
```

Comment:
What is meant by transparent to the user here?

Reply:
I'm not sure the `printf`s are useful, to be honest: people still have to read the code to work out what's going on. I'm generally in favour of programs outputting the bare minimum needed for comprehension.

Reply:
Yes, I have mixed feelings here. What do you suggest? Removing all the `ptintf`s?

Reply:
If they don't tell us something important, then I would get rid of them.